eox_nelp/notifications/tests/test_notify_course_due_date.py

```diff
@@ -100,39 +91,113 @@ def test_notify_upcoming_course_due_date(self):
 
         with self.assertLogs(notify_course_due_date.__name__, level="INFO") as logs:
             notify_upcoming_course_due_date(instance)
 
-        self.course_email_template_mock.get_template().render_plaintext.assert_called_with(
-            expected_text_msg,
-            expected_email_context,
+        self.course_email_template_mock.get_template().render_plaintext.assert_called_with("", expected_email_context)
+        self.course_email_template_mock.get_template().render_htmltext.assert_called_with("", expected_email_context)
+        self.send_email_multialternative_mock.assert_called_with(
+            subject=expected_subject,
+            plaintext_msg=expected_text_msg,
+            html_msg=expected_html_msg,
+            recipient_emails=recipients_list,
         )
-        self.course_email_template_mock.get_template().render_htmltext.assert_called_with(
-            expected_html_msg,
-            expected_email_context,
+        self.assertTrue(instance.sent)
+        succesful_log = f"{sent_emails} emails sent related the upcoming_course_due_date: {instance.__dict__} -------"
+        self.assertEqual(logs.output, [f"INFO:{notify_course_due_date.__name__}:{succesful_log}"])
+
+    @override_settings()
+    def test_notify_upcoming_course_due_date_custom_subject(self):
+        """Test correct behaviour of sending email with
+        method `notify_upcoming_course_due_date`.
+        Expected behavior:
+            - course_email_template.render_plaintext is called with the right values.
+            - course_email_template.render_htmltext is called with the right values.
+            - send_email_multialternative is called with the right values.
+            - The email was sent from the correct email address (if specified)
+            - sent attribute of instance of `UpcomingCourseDueDate` after the process is True.
+            - Log successful of sent message.
+        """
+        subsection_due_date = timezone.now() + timedelta(days=4)
+        instance, _ = UpcomingCourseDueDate.objects.update_or_create(  # pylint: disable=no-member
+            course=CourseOverview.objects.create(id="course-v1:test+Cx105+2022_T4"),
+            location_id="block-v1:test+CS501+2022_T4+type@sequential+block@a54730a9b89f420a8d0343dd581b447a",
+            due_date=subsection_due_date,
+            notification_date=timezone.now(),
+            sent=False,
+            defaults={"sent": False},
         )
-        self.send_email_multialternative_mock(
-            expected_subject,
-            expected_text_msg,
-            expected_html_msg,
-            recipients_list,
+        course_title = "course-display-name"
+        subsection_title = "subsection-display-name"
+        recipients_list = ["vader@star.com", "palpatine@star.com"]
+        subsection_delta = subsection_due_date - timezone.now()
+        expected_text_msg = "expected-text-msg"
+        expected_html_msg = "expected-html-msg"
+        setattr(
```

Comment:
I think it's better to use the decorator in order to revert this change after the test

Reply:
but I am usinghttps://github.com/eduNEXT/eox-nelp/blob/0cbd232bf88f915eda14ed9ec76c4c5e1f823b70/eox_nelp/notifications/tests/test_notify_course_due_date.py#L126 @override settings decorator? is for this?

Reply:
I didn't realize about it, since normally the decorator includes the new value